kili/mutations/label/__init__.py

```diff
@@ -59,12 +61,18 @@ def create_predictions(
         if len(external_id_array) == 0:
             warnings.warn("Empty IDs and prediction list")
 
-        variables = {
-            'data': {'modelNameArray': model_name_array,
-                     'jsonResponseArray': [dumps(elem) for elem in json_response_array]},
-            'where': {'externalIdStrictlyIn': external_id_array, 'project': {'id': project_id}}
-        }
-        result = self.auth.client.execute(GQL_CREATE_PREDICTIONS, variables)
+        for batch_number, (external_id_array_batch, model_name_array_batch, json_response_array_batch)\
+                in enumerate(batch_iterators_builder([external_id_array, model_name_array, json_response_array])):
```

Comment:
batch_iterators_builder seems to be the first step at abstracting away the logic, but you should abstract away the rest. We can do a technical strategy next week for that.